ddroberts77/stock-trading-ai
# Fix test requirements and update test files

This PR fixes the test requirements and updates the test files to address the failing tests. Changes include:

1. Updated requirements.txt:
- Added proper testing dependencies (pytest, pytest-cov, pytest-xdist)
- Added core dependencies (numpy, pandas, scikit-learn, torch)
- Added development tools (black, isort, flake8)

2. Updated test files:
- Fixed device handling in PyTorch tests
- Added proper test fixtures
- Improved test coverage and assertions
- Added gradient flow testing
- Removed TensorFlow dependencies

3. Test Improvements:
- Added proper error handling
- Added numerical stability checks
- Added shape and value validation
- Added gradient checks for model training

To implement these changes:
1. Pull this branch
2. Run `pip install -r requirements.txt`
3. Run `pytest tests/ -v` to verify the changes

The changes ensure that tests will run consistently across Python 3.9, 3.10, and 3.11 environments.
